src/spec/controllers/pages_controller_spec.rb

```diff
@@ -1,19 +1,33 @@
 require 'spec_helper'
 
-describe PagesController do
+RSpec.describe PagesController do
   context "with an event" do
     let!(:event) { create(:event, :full_event) }
+    let!(:scheduled_break) do
+       create(:session,
+              title: 'Break time',
+              event: event,
+              room: event.rooms.first,
+              participant: create(:participant),
+              timeslot: event.timeslots.first )
+    end
+    let!(:proposed_session) do
+       create(:session, title: 'How to break things', event: event, room: nil)
+    end
 
-    it "should show the sessions" do
+    it "shows the sessions" do
       get :home
       expect(response).to be_successful
-      expect(assigns[:recent_sessions]).to be_empty
+      # We are implicitly testing that scheduled_break is not in either set
+      expect(assigns[:recent_sessions]).to eq [proposed_session]
+      expect(assigns[:random_sessions]).to eq [proposed_session]
```

Comment:
I was expecting to see one of these cover the case when the timeslot is nil... Can you add that (or did I miss it?)

Reply:
Timeslot is nil on `proposed_session`